src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -1353,28 +1388,152 @@ void CodegenNeuronCppVisitor::print_nrn_state() {
+        std::string sum;
+        for (const auto& current: info.currents) {
+            auto var = breakpoint_current(current);
+            sum += get_variable_name(var);
+            if (&current != &info.currents.back()) {
+                sum += "+";
+            }
+        }
+        printer->fmt_line("double rhs = {};", sum);
+    }
+
+    std::string sum;
+    for (const auto& conductance: info.conductances) {
+        auto var = breakpoint_current(conductance.variable);
+        sum += get_variable_name(var);
+        if (&conductance != &info.conductances.back()) {
+            sum += "+";
+        }
+    }
+    printer->fmt_line("double g = {};", sum);
+
+    for (const auto& conductance: info.conductances) {
+        if (!conductance.ion.empty()) {
+            const auto& lhs = std::string(naming::ION_VARNAME_PREFIX) + "di" + conductance.ion +
+                              "dv";
+            const auto& rhs = get_variable_name(conductance.variable);
+            const ShadowUseStatement statement{lhs, "+=", rhs};
+            const auto& text = process_shadow_update_statement(statement, BlockType::Equation);
+            printer->add_line(text);
+        }
+    }
 }
 
 
 /// TODO: Edit for NEURON
 void CodegenNeuronCppVisitor::print_nrn_cur_non_conductance_kernel() {
-    return;
+    printer->fmt_line("double I1 = nrn_current_{}({}+0.001);",
+                      info.mod_suffix,
+                      nrn_current_arguments());
+    for (auto& ion: info.ions) {
+        for (auto& var: ion.writes) {
+            if (ion.is_ionic_current(var)) {
+                const auto& name = get_variable_name(var);
+                printer->fmt_line("double di{} = {};", ion.name, name);
+            }
+        }
+    }
+    printer->fmt_line("double I0 = nrn_current_{}({});", info.mod_suffix, nrn_current_arguments());
+    printer->add_line("double rhs = I0;");
+
+    printer->add_line("double g = (I1-I0)/0.001;");
+    for (auto& ion: info.ions) {
+        for (auto& var: ion.writes) {
+            if (ion.is_ionic_current(var)) {
+                const auto& lhs = std::string(naming::ION_VARNAME_PREFIX) + "di" + ion.name + "dv";
+                auto rhs = fmt::format("(di{}-{})/0.001", ion.name, get_variable_name(var));
+                if (info.point_process) {
+                    auto area = get_variable_name(naming::NODE_AREA_VARIABLE);
+                    rhs += fmt::format("*1.e2/{}", area);
+                }
+                const ShadowUseStatement statement{lhs, "+=", rhs};
+                const auto& text = process_shadow_update_statement(statement, BlockType::Equation);
+                printer->add_line(text);
+            }
+        }
+    }
 }
 
 
 /// TODO: Edit for NEURON
 void CodegenNeuronCppVisitor::print_nrn_cur_kernel(const BreakpointBlock& node) {
-    return;
+    printer->add_line("int node_id = node_data.nodeindices[id];");
+    printer->add_line("double v = node_data.node_voltages[node_id];");
```

Comment:
Just a note: In the nmodl-references I asked about the usage and I see how they used.

Reply:
I've marked the location for anyone else :)